File: BaekJoon/Stack/10773.py
import sys


N = int(sys.stdin.readline())
stack = []
for _ in range(N):
    num = int(sys.stdin.readline())
    if  num  != 0:
        stack.append(num)
    else:
        stack.pop()

result = 0

for n in stack:
    result+=n

print(result)

File: BaekJoonSwift/BaekJoonSwift/Stack (1)/10773.swift
import Foundation

let n = Int(readLine()!)!
var result = 0
var stack = [Int]()

for _ in 0..<n {
    let temp = Int(readLine()!)!
    if temp == 0 {
        result -= stack.removeLast()
    }
    else {
        stack.append(temp)
        result += temp
    }
}
print(result)
